[REF] Remove usages of CRM_Utils_Money::format that pass in a blank s…

seamuslee001 opened the pull request:
…pace as a currency

Overview
----------------------------------------
This is a reviewers cut of https://github.com/civicrm/civicrm-core/pull/18014 and but does things slightly differently, in that it doesn't add in the deprecation as that needs to be dealt with once we bring in brick money, it also adds in a new parameter to determine if when we are returning just the number whether to replace the decimal and thousands separators

Before
----------------------------------------
Error messages such as 
![campaign_error_pre](https://user-images.githubusercontent.com/6799125/90206220-7ae25c00-de26-11ea-9473-bcc694e02aba.jpg)
![contribution_query_pre](https://user-images.githubusercontent.com/6799125/90206222-7f0e7980-de26-11ea-9ef5-b0b644de9ee7.jpg)
![contribution_page_pre_settings](https://user-images.githubusercontent.com/6799125/90206224-833a9700-de26-11ea-8d6e-d11decdf2d35.jpg)
and form is prefilled like so:
![contribution_page_amount_pre](https://user-images.githubusercontent.com/6799125/90206248-8afa3b80-de26-11ea-8abf-8555ff4aeed4.jpg)

After
----------------------------------------
No change to the way the error message shows or how the form is prefilled
![campaign_error_post](https://user-images.githubusercontent.com/6799125/90206271-9a798480-de26-11ea-94ec-2cf9d8449e8a.jpg)
![contribution_query_post](https://user-images.githubusercontent.com/6799125/90206276-9ea5a200-de26-11ea-89f6-a88338c9747b.jpg)
![contribution_page_post_settings](https://user-images.githubusercontent.com/6799125/90206283-a2392900-de26-11ea-8566-6912d82ab927.jpg)
![contribution_page_amount_post_us_settings](https://user-images.githubusercontent.com/6799125/90206287-a5341980-de26-11ea-9fc9-5fe12916752f.jpg)
![contribution_page_amount_post_eu_settings](https://user-images.githubusercontent.com/6799125/90206294-a82f0a00-de26-11ea-8036-5b16fd1a3248.jpg)

ping @eileenmcnaughton @mattwire @totten

eileenmcnaughton commented on CRM/Utils/Money.php:
```diff
@@ -44,12 +44,15 @@ class CRM_Utils_Money {
    * @param bool $onlyNumber
    * @param string $valueFormat
    *   The desired monetary value display format (e.g. '%!i').
+   * @param bool $onlyNumberReplaceSeparators
```
Dang - this just highlights to me that we should have a separate function rather than this big overload - don't we have one?

seamuslee001 replied:
Nope we don't because the replacement only occurs in this function and only if we have not specified to return just the number

eileenmcnaughton replied:
@seamuslee001 I gave this a quick test - https://github.com/civicrm/civicrm-core/pull/18151

seamuslee001 commented:
@mattwire @eileenmcnaughton as Eileen found in #18151 there is already a function to do what we want here so I have updated everything except for the campaign one to use the new function. For the campaign one I have reverted the change I had made in this PR as that will be covered by Eileen's PR #18151
